### DISCUSSION TITLE
AWS CloudFront Server-Timing header

### DISCUSSION THREAD

**Discussion by MrBazzieB** [aws/aws-cli#6868] [Q&A] [answered]
Hi there

`aws-cli/2.5.4 Python/3.9.12 Darwin/21.4.0 source/x86_64 prompt/off`

With the release of the [Server Timing headers](https://aws.amazon.com/blogs/networking-and-content-delivery/amazon-cloudfront-introduces-server-timing-headers/) for CloudFront, when would this be updated for the AWS CLI? When running 
`aws cloudfront list-response-headers-policies --type custom ` It does not return the server-timing header setting at all. Below is a basic response header policy with only server-timing header enabled:

`{
                "Type": "custom",
                "ResponseHeadersPolicy": {
                    "Id": "REDACTED",
                    "LastModifiedTime": "REDACTED",
                    "ResponseHeadersPolicyConfig": {
                        "Comment": "",
                        "Name": "server-timing-enabled",
                        "SecurityHeadersConfig": {
                            "XSSProtection": {},
                            "FrameOptions": {},
                            "ReferrerPolicy": {},
                            "ContentSecurityPolicy": {},
                            "ContentTypeOptions": {},
                            "StrictTransportSecurity": {}
                        }
                    }
                }
            }`

I do believe I am on the latest CLIv2 release.

**Comment by tim-finnigan** [comment, marked as answer]
Hi @MrBazzieB if I enable Server Timing headers and add `--debug` to that command then I see the `ServerTimingHeadersConfig` value returned in the logs. I confirmed that is included in the underlying API [request syntax](https://docs.aws.amazon.com/cloudfront/latest/APIReference/API_ListResponseHeadersPolicies.html). It just has to be modeled upstream by the service team so that the CLI/SDKs can return it in the response. I created an issue [here](https://github.com/aws/aws-sdk/issues/223) for the Cloudfront team and will share an update when I hear back. Thanks!
